### DISCUSSION TITLE
maximum size of canvas?

### DISCUSSION THREAD

**Discussion by saschamarco** [xournalpp/xournalpp#5507] [Q&A]
If the canvas gets too big the zoom-levels are restricted, for example at 3840x2160 "points" i can no longer zoom in all the way to 700%, it will only display a black slate then.
How does that work and is there a way around it?

Thx
sascha

**Comment by rolandlo** [comment]
There is mistery around it, see #5333, which we don't understand yet.

**Comment by bhennion** [comment]
Not sure it is the same as #5333 (which is weird): here the issue is our rendering system: one page is one surface, so a 3840*2160 points page at 700% would mean a 1.5GB cairo_surface allocated. The library is just not made for this stupidly big surface.

There is currently no workaround. Implementing a tiles based rendering is a long standing objective, but not to be expected any time soon.
